lib/rails_erd/diagram.rb

```diff
@@ -138,6 +146,27 @@ def generate
       end
     end
 
+    def recurse_into_relationships(entity, max_level, current_level = 0)
+      return [] unless entity
+      return [] if max_level == current_level
+
+      relationships = entity.relationships.reject(&:recursive?)
+      relationships = relationships.reject(&:indirect?)
```

Comment:
I think that's probably fine. What do you think about collapsing these 2 lines into 

```
relationships = entity.relationships.reject{ |r|  r.indirect? || r.recursive? }
```

That would let us only have to scan through once (and placing the `indirect` first, since it'll likely fail more often than `recursive`).